src/main/java/org/prebid/server/bidder/kubient/KubientBidder.java

```diff
@@ -0,0 +1,117 @@
+package org.prebid.server.bidder.kubient;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.EncodeException;
+import io.vertx.core.json.Json;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * kubient {@link Bidder} implementation.
+ */
+public class KubientBidder implements Bidder<BidRequest> {
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+
+    public KubientBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(endpointUrl);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+
+        String body;
+        try {
+            body = Json.encode(request);
+        } catch (EncodeException e) {
+            errors.add(BidderError.badInput(String.format("Failed to encode request body, error: %s", e.getMessage())));
+            return Result.of(Collections.emptyList(), errors);
```

Comment:
Since there can be only one error, no need to introduce `errors` list and manually add error to it - instead, you can instantly return `Result.emptyWithError(BidderError.badInput(...))` in catch block. For normal Result return - just use an emptyList() for errors.